lib/features/allow_ptrace.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * This feature adjusts the AppArmor profile to allow ptrace within a container.
+ * All of the interesting action takes place in `lib/task.js`.
+ */
+
+export default class AllowPtrace {
+  constructor() {
+    this.featureName = 'allowPtrace';
```

Comment:
Also, if this should be something protected behind a scope, we can do what testdroid or balrog proxy do now as the burden of refactoring how the scopes are checked shouldn't be your problem but I would like to get a bug entered so I can move that feature scope checking out of each individual class and have something before that when the features are first initialized.

Perhaps call a method on each feature here to see if the task is authorized to use that feature:
https://github.com/taskcluster/docker-worker/blob/master/lib/task.js#L65